gen/list.go

```diff
@@ -140,6 +140,60 @@ func (l *listGenerator) Reader(g Generator, spec *compile.ListSpec) (string, err
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
 
+// Decoder generates a function to read a list of the given type from a
+// stream.Reader.
+//
+//     func $name(sr *stream.Reader) ($listType, error) {
+//             ...
+//     }
+//
+// And returns its name.
+func (l *listGenerator) Decoder(g Generator, spec *compile.ListSpec) (string, error) {
+	name := decoderFuncName(g, spec)
+	err := g.EnsureDeclared(
+		`
+		<$stream := import "go.uber.org/thriftrw/protocol/stream">
+		<$listType := typeReference .Spec>
+
+		<$sr := newVar "sr">
+		<$lh := newVar "lh">
+		<$o := newVar "o">
+		<$v := newVar "v">
+		func <.Name>(<$sr> <$stream>.Reader) (<$listType>, error) {
+			<$lh>, err := <$sr>.ReadListBegin()
+			if err != nil {
+				return nil, err
+			}
+
+			// TODO(wit): Should there be validation here between the expected list's
+			// type and the type of .Spec.ValueSpec?  The validation will error,
+			// unlike non-streaming behavior.
+
+			<$o> := make(<$listType>, 0, <$lh>.Length)
+			for i := 0; i != <$lh>.Length; i++ {
```

Comment:
> `i != <$lh>.Length`

I did this really hacky thing for now - the generator was complaining about the `<` symbol:
```
unexpected "<" in command
```

Reply:
Resolved now that #499 fixes this issue.